budget app backend/controllers/expanse_controller.js

```diff
@@ -1,35 +1,96 @@
+const transactions = require("../models/transactions");
+
 module.exports = class {
-    static get_all_expanse(req, res) {
-        res.status(200).json({
-            ok: true,
-            message: `All expanse GET!`
+    static async get_all_expanse(req, res) {
+        let results = await transactions.find({
+            account_id: req.body.account_id,
+            type: "expanse"
         })
-    }
-    static get_expanse_by_id(req, res) {
         res.status(200).json({
             ok: true,
-            message: `${req.params.id} expanse has been requested`
+            results,
+            message: `All expanse for ${req.body.account_id}!`
         })
     }
-    static create_expanse(req, res) {
-        const { title } = req.body; //there will be other fields, like expanse_amount
-        res.status(201).json({
-            ok: true,
-            message: `${title} expanse has been created`
-        })
+    static async get_expanse_by_id(req, res) {
+        const result = await transactions.findById(req.params.id)
+        if (result) {
+            res.status(200).json({
+                ok: true,
+                result,
+                message: `${req.params.id} expanse has been requested`
+            })
+        } else {
+            res.status(403).json({
+                ok: false,
+                message: `${req.params.id} expanse doesn't exist`
+            })
+        }
+    }
+    static async create_expanse(req, res) {
+        const { title, amount, description, payer, pay_day, category_id, account_id } = req.body;
+        try {
+            const created = await transactions.create({
```

Comment:
Why dont pass body as parameter? 

`await transactions.create(req.body)`

Reply:
I corrected them. Thanks for suggestions